docs/src/pages/getting-started/astro.mdx

```diff
@@ -1 +1,229 @@
+
+const auth = (req: Request) => ({ id: "fakeId" }); // Fake auth function
+
+// FileRouter for your app, can contain multiple FileRoutes
+export const ourFileRouter = {
+  // Define as many FileRoutes as you like, each with a unique routeSlug
+  imageUploader: f({ image: { maxFileSize: "4MB" } })
+    // Set permissions and file types for this FileRoute
+    .middleware(async ({ req }) => {
+      // This code runs on your server before upload
+      const user = await auth(req);
+
+      // If you throw, the user will not be able to upload
+      if (!user) throw new Error("Unauthorized");
+
+      // Whatever is returned here is accessible in onUploadComplete as `metadata`
+      return { userId: user.id };
+    })
+    .onUploadComplete(async ({ metadata, file }) => {
+      // This code RUNS ON YOUR SERVER after upload
+      console.log("Upload complete for userId:", metadata.userId);
+
+      console.log("file url", file.url);
+
+      // !!! Whatever is returned here is sent to the clientside `onClientUploadComplete` callback
+      return { uploadedBy: metadata.userId };
+    }),
+} satisfies FileRouter;
+
+export type OurFileRouter = typeof ourFileRouter;
+```
+
+### Create a Astro API route using the FileRouter
+
+<Callout type="info">
+  File path here doesn't matter, you can serve this from any route. We recommend
+  serving it from `/api/uploadthing`.
+</Callout>
+
+```ts copy filename="src/pages/api/uploadthing.ts"
+import { createRouteHandler } from "uploadthing/server";
+
+import { ourFileRouter } from "~/server/uploadthing";
+
+// Export routes for Next App Router
+export const { GET, POST } = createRouteHandler({
+  router: ourFileRouter,
+  config: {
+    uploadthingSecret: import.meta.env.UPLOADTHING_SECRET,
+  },
+});
+```
+
+> See configuration options in
+> [server API reference](/api-reference/server#createroutehandler)
+
+### Add UploadThing's Styles
+
+<Tabs items={["Tailwind", "Not Tailwind"]}>
+  <Tab>
+    <TailwindSetup />
+  </Tab>
+  <Tab>
+    Include our CSS file in the root layout to make sure the components look right!
+
+    ```astro copy filename="src/layouts/root.astro"
+    ---
+    import "@uploadthing/react/styles.css";
+    // ...
+    ---
+    ```
+
+  </Tab>
+</Tabs>
+
+### Create The UploadThing Components (Recommended)
+
+We provide components to make uploading easier. We **highly recommend
+re-exporting them with the types assigned**, but you CAN import the components
+individually from `@uploadthing/react` instead.
+
+```ts copy filename="src/utils/uploadthing.ts"
+import {
+  generateUploadButton,
+  generateUploadDropzone,
+} from "@uploadthing/react";
+
+import type { OurFileRouter } from "~/server/uploadthing";
+
+export const UploadButton = generateUploadButton<OurFileRouter>();
+export const UploadDropzone = generateUploadDropzone<OurFileRouter>();
+```
+
+### Mount A Button And Upload!
+
+<Callout type="info">
+  Don't forget to add the `"use client;"` directive at the top of your file,
```

Comment:
Is this needed in Astro? Should we be telling them to use the client directives on the Astro side?

Reply:
whoops nope